### PR TITLE
[VL-104] Add GetNotificationDocumentMetadata endpoint

### PR DESCRIPTION
**Pull request by datalek**
#### Motivation and Context
Add a new endpoint that returns the metadata of a notified document

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/domain/GetNotificationDocumentMetadataRepository.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { Repository } from './Repository';
+import { Response, UnauthorizedMessageBody } from './types';
+import { Notification } from './Notification';
+
+export type GetNotificationDocumentMetadataRecord = {
+  type: 'GetNotificationDocumentMetadataRecord';
+  input: { apiKey: ApiKey; iun: Iun; docIdx: number };
+  output:
+    | Response<200, NotificationAttachmentDownloadMetadataResponse>
+    | Response<403, UnauthorizedMessageBody>
```
Comment (kin0992) [inline src/domain/GetNotificationDocumentMetadataRepository.ts:13]: I leave this comment here, just as a reminder and we can discuss about it in an appropriate meeting.

It looks like that now, if you make a request without the `x-api-key` header, the server returns a 401; if you make a call with the header, but with an invalid value, the server returns 403.

Here an example of a request that returns a 401:
```
curl --location --request GET 'https://api.dev.pn.pagopa.it/delivery/notifications/sent/6002032.770410135/attachments/documents/1' \
--header 'Accept: application/json'
```

Here an example of a request that returns a 403:
```
curl --location --request GET 'https://api.dev.pn.pagopa.it/delivery/notifications/sent/6002032.770410135/attachments/documents/1' \
--header 'Accept: application/json' \
--header 'x-api-key: asd'
```

File: src/useCases/GetNotificationDocumentMetadataUseCase.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import crypto from 'crypto';
+import { flow, pipe } from 'fp-ts/lib/function';
```
Comment (kin0992) [inline src/useCases/GetNotificationDocumentMetadataUseCase.ts:2]:
```suggestion
import { flow, pipe } from 'fp-ts/function';
```

File: src/useCases/__tests__/GetNotificationDocumentMetadataUseCase.test.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import { makeLogger } from '../../logger';
+import { NewNotificationRecord } from '../../domain/NewNotificationRepository';
+import { CheckNotificationStatusRecord } from '../../domain/CheckNotificationStatusRepository';
+import { ConsumeEventStreamRecord } from '../../domain/ConsumeEventStreamRecordRepository';
+import { GetNotificationDocumentMetadataRecord } from '../../domain/GetNotificationDocumentMetadataRepository';
+
+const logger = makeLogger();
+const occurrencesAfterComplete = 2;
+
+describe('GetNotificationStatusUseCase', () => {
```
Comment (kin0992) [inline src/useCases/__tests__/GetNotificationDocumentMetadataUseCase.test.ts:14]:
```suggestion
describe('GetNotificationDocumentMetadataUseCase', () => {
```

**Review by kin0992 — CHANGES_REQUESTED** [review]

File: src/domain/GetNotificationDocumentMetadataRepository.ts (hunk 1 above)
Comment (datalek) [reply src/domain/GetNotificationDocumentMetadataRepository.ts:13]: We should open a ticker about this mismatch

**Review by kin0992 — APPROVED** [review]

File: src/domain/GetNotificationDocumentMetadataRepository.ts (hunk 1 above)
Comment (kin0992) [reply src/domain/GetNotificationDocumentMetadataRepository.ts:13]: This activity is tracked with [this](https://pagopa.atlassian.net/browse/VL-115) issue.

**Pull request merged**